fuel-core/src/database/vm_database.rs

```diff
@@ -162,79 +169,46 @@ impl InterpreterStorage for VmDatabase {
         start_key: &Bytes32,
         range: Word,
     ) -> Result<Vec<Option<Cow<Bytes32>>>, Self::DataError> {
+        // TODO: Optimization: Iterate only over `range` elements.
         let mut iterator = self.database.iter_all::<Vec<u8>, Bytes32>(
```

Comment:
`iter_all` returns only entries with the same `prefix` by default. We don't need to check it

Reply:
True, this should be covered by tests at the DB level against whatever backends it supports since the interface should guarantee the prefix is always correct.